templates/index.html

```diff
@@ -6,12 +6,16 @@
     <script async src="https://cdn.ampproject.org/v0.js"></script>
     <script async custom-element="amp-sidebar" src="https://cdn.ampproject.org/v0/amp-sidebar-0.1.js"></script>
     <script async custom-element="amp-selector" src="https://cdn.ampproject.org/v0/amp-selector-0.1.js"></script>
+    <script async custom-element="amp-consent" src="https://cdn.ampproject.org/v0/amp-consent-0.1.js"></script>
```

Comment:
Can you move the imports to head.html. And, this is where it gets ugly, you need to surround it with checks if the sample already contains the imports to avoid duplicate imports:

```
{{^includesGeo}}
<script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"></script>
{{/includesAmpGeo}}
{{^includesConsent}}
<script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"></script>
{{/includesAmpConsent}}
```

The variables need to be defined here: https://github.com/ampproject/amp-by-example/blob/master/tasks/compile-example.js#L231

Reply:
This might get a bit weird though for the amp-consent samples, it might make sense to disable the dialog there.

Reply:
Ok, done.

I'd wanted for a while to write something that simply detects which components are used on an AMP and inserts the proper `<script>`s.  So @garanj kindly did this for our new e-comm project:

https://github.com/morsssss/amp-ecomm/pull/3

I get what you say about confusion on the amp-consent samples - but I'd just guess that legally we're required to have this cookie notice sitewide.

Reply:
We have to see, if it works and validates it's fine to keep it. Otherwise I'd say we have to remove it from the page.

Reply:
It looks fine on the amp-consent samples... but there are indeed errors as the pages have two amp-consent's.  And it doesn't appear on the geolocation-based page at all.  That is, `user_consent/geolocation-based_consent_flow`.

Dang!

@rudygalfi, @ericlindley-g - any opinion here?

Reply:
I would say that if possible we should have the pages validate and have the notice site-wide. Is there an option that does both? In the sample can we just point to code that actually implements the consent for the page, rather than a faked example?